File: algorithms/strings/gem-stones/gem-stones-javascript/js/testeGemstones.js


import {gemstones} from '../js/gemstones.js';
import assert from 'assert';

describe('gemstones', function() {

    it('O número de diferentes tipos de elementos-gema que ele tem é: 2', function(){

        let arr = ['abcdde', 'baccd', 'eeabg'];
        
        let esperado = 2;
        
        let resultado = gemstones(arr);

        assert.strictEqual(resultado, esperado);

    });
});

describe('gemstones', function() {

    it('O número de diferentes tipos de elementos-gema que ele tem é: 4', function(){

        let arr = ['basdfj', 'asdlkjfdjsa', 'bnafvfnsd', 'oafhdlasd'];
        
        let esperado = 4;
        
        
        let resultado = gemstones(arr);

        assert.strictEqual(resultado, esperado);

    });
});


describe('gemstones', function() {

    it('O número de diferentes tipos de elementos-gema que ele tem é: 0', function(){

        let arr = ['vtrjvgbj', 'mkmjyaeav', 'sibzdmsk'];
        
        let esperado = 0;
        
        let resultado = gemstones(arr);

        assert.strictEqual(resultado, esperado);

    });
});

File: algorithms/strings/gem-stones/gem-stones-java/gem-stones-java/src/TesteGemstones.java
import static org.junit.Assert.assertEquals;

import java.util.*;

import org.junit.Test;

public class TesteGemstones {
    
    @Test
    public void testeCaso1() {

        List<String> arr = Arrays.asList("abcdde", "baccd", "eeabg");

        int esperado = 2;
        
        int resultado = Resultado.gemstones(arr);

        assertEquals(esperado, resultado);
    }

    @Test
    public void testeCaso2() {

        List<String> arr = Arrays.asList("basdfj", "asdlkjfdjsa", "bnafvfnsd", "oafhdlasd");

        int esperado = 4;
        
        int resultado = Resultado.gemstones(arr);

        assertEquals(esperado, resultado);
    }

    @Test
    public void testeCaso3() {

        List<String> arr = Arrays.asList("vtrjvgbj", "mkmjyaeav", "sibzdmsk");

        int esperado = 0;
        
        int resultado = Resultado.gemstones(arr);

        assertEquals(esperado, resultado);
    }
}
